spec/spec_helper.rb

```diff
@@ -43,8 +43,14 @@ def reset_all
   def with_unfulfilled_double
     d = double("double")
     yield d
+  ensure
     reset d
   end
+
+  def expect_fast_failure_from(double, class_or_message=Exception, message=/.*/, &blk)
```

Comment:
`Exception` seems like a poor choice for the default here, given that at virtually all the call sites above you pass `RSpec::Mocks::MockExpectationError` (and in the places you don't, I'm wondering if you should!).  It seems like the norm is `RSpec::Mocks::MockExpectationError` -- after all, in rspec-mocks, that's who we signal failures -- so we should make that the default and only make the caller pass a different class if it really needs something different.